actors/util/adt/balancetable.go

```diff
@@ -98,17 +74,17 @@ func (t *BalanceTable) SubtractWithMinimum(key addr.Address, req abi.TokenAmount
 	return sub, nil
 }
 
-// MustSubtract subtracts the given amount from the account's balance
-// returns an error if account isn't created or if it has insufficient balance
+// MustSubtract subtracts the given amount from the account's balance.
+// Returns an error if the account has insufficient balance
 func (t *BalanceTable) MustSubtract(key addr.Address, req abi.TokenAmount) error {
-	subst, err := t.SubtractWithMinimum(key, req, big.Zero())
+	prev, err := t.Get(key)
 	if err != nil {
 		return err
 	}
-	if !subst.Equals(req) {
-		return errors.New("couldn't subtract the requested amount")
+	if req.GreaterThan(prev) {
+		return xerrors.New("couldn't subtract the requested amount")
 	}
-	return nil
+	return t.Add(key, req.Neg())
```

Comment:
This can end up storing zero. Should we delete in this case?

Reply:
Yes!